mllib/src/test/scala/org/apache/spark/ml/classification/FMClassifierSuite.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.classification
+
+import org.apache.spark.ml.classification.LogisticRegressionSuite.generateLogisticInput
+import org.apache.spark.ml.linalg.{DenseVector, Vector, Vectors}
+import org.apache.spark.ml.param.ParamsSuite
+import org.apache.spark.ml.regression.FMRegressorSuite._
+import org.apache.spark.ml.util._
+import org.apache.spark.ml.util.TestingUtils._
+import org.apache.spark.sql.{DataFrame, Row}
+
+class FMClassifierSuite extends MLTest with DefaultReadWriteTest {
```

Comment:
Can we compare the fitted coefficients with libFM?

Reply:
I want to compare with libFM, but randomly generate a binary dataset, FM can't fit the dataset perfectly. Generally, I get very small loss value, but the parameters are not exactly same every time (also in libFM).

My current test method is that for mse to generate a feature-interaction dataset, FMRegressor needs to fit this dataset perfectly, bias/linearVector needs same, every `<v_i, v_j>` needs same that calculates by factorMatrix.

However, for FMClassifier, it is difficult to generate a binary feature-interaction dataset, so for the FMClassifier, I don't verify the correctness of interaction logic (that is verify in mse), only verify the correctness of logloss logic.

Reply:
I still doubt whether existing testsuite is enough. 
1, I suggest to add several testsuites to check whether `FMClassifier` and `FMRegressor` can learn the intercept correctly. 
Like '// Test if we can correctly learn Y = 0.1 + 1.2*X1 - 1.3*X2 + 20*X1*X2' in https://github.com/apache/spark/pull/5591/files

2, I suggest to add some testcases with `fitBias==false` or/and `fitLinear==false`

Reply:
Is this resolved? I'd also love to have at least one test that compares output to some other library, even if the test can only assert some approximate match in loss, etc.

Reply:
> Is this resolved? I'd also love to have at least one test that compares output to some other library, even if the test can only assert some approximate match in loss, etc.

Now, test method is FM fit a dataset perfectly (loss closed to zero and model's weights closed to real weights), the dataset is generated by a group weight (include bias/linear/factor weights). I think it is enough.